packages/components/ovh-shell/src/plugin/ux/index.ts

```diff
@@ -0,0 +1,82 @@
+import Shell from '../../shell/shell';
+import { ShellUX } from './ux';
+
+export interface IUXPlugin {
+  isAccountSidebarVisible(sidebarName: string): boolean;
+  showAccountSidebar(disableToggle: boolean): void;
+  hideAccountSidebar(): void;
+  toggleAccountSidebarVisibility(): void;
+  isNotificationsSidebarVisible(): boolean;
+  showNotificationsSidebar(): void;
+  hideNotificationsSidebar(): void;
+  enableAccountSidebarVisibilityToggle(): void;
+  disableAccountSidebarVisibilityToggle(): void;
+  toggleNotificationsSidebarVisibility(): void;
+  toggleAccountSidebarVisibility(): void;
+}
+
+// TODO: remove this once we have a more generic Plugin class
+export type UXPluginType<T extends UXPlugin> = {
+  [key in keyof T]?: T[key];
+};
+
+export class UXPlugin implements IUXPlugin {
+  private shellUX: ShellUX;
+
+  constructor(shell: Shell) {
+    this.shellUX = new ShellUX(shell);
+    this.shellUX.registerSidebar('account');
+    this.shellUX.registerSidebar('notifications');
+    this.shellUX.registerNavbar();
+  }
+
+  /* ----------- AccountSidebar methods -----------*/
+
+  isAccountSidebarVisible(): boolean {
+    return this.shellUX.isSidebarVisible('account');
+  }
+
+  showAccountSidebar(disableToggle = false): void {
```

Comment:
it's really nitpicking here but adding something like :
```js
lockAccountSidebar()
```
would be even better because when you are reading :
```js
showAccountSidebar(true)
```
the role of the boolean parameter is not obvious

Reply:
I agree, will change !